src/plan.rs

```diff
@@ -0,0 +1,139 @@
+use std::{
+    cmp::Ordering,
+    collections::{BinaryHeap, HashMap},
+};
+
+pub struct Id {
+    id: usize,
+}
+
+pub struct Plan<T> {
+    pub time: f64,
+    pub data: T,
+}
+
+#[derive(PartialEq, Debug)]
+pub struct Record {
```

Comment:
```suggestion
struct Record {
```

I would also consider calling this `QueueEntry` or `Entry`